tsl/test/expected/continuous_aggs_refresh.out

```diff
@@ -371,3 +371,68 @@ ORDER BY 1,2;
 
 -- Try max refresh window size
 CALL refresh_continuous_aggregate('cond_20_bigint', NULL, NULL);
+-- Test that WITH NO DATA and WITH DATA works (we use whatever is the
+-- default for Postgres, so we do not need to have test for the
+-- default).
+CREATE MATERIALIZED VIEW weekly_temp_no_refresh
+WITH (timescaledb.continuous,
+      timescaledb.materialized_only=true)
+AS
+SELECT time_bucket('7 days', time) AS day, device, avg(temp) AS avg_temp
+FROM conditions
+GROUP BY 1,2 WITH NO DATA;
+NOTICE:  adding index _materialized_hypertable_11_device_day_idx ON _timescaledb_internal._materialized_hypertable_11 USING BTREE(device, day)
+CREATE MATERIALIZED VIEW weekly_temp_refresh
+WITH (timescaledb.continuous,
+      timescaledb.materialized_only=true)
+AS
+SELECT time_bucket('7 days', time) AS day, device, avg(temp) AS avg_temp
+FROM conditions
+GROUP BY 1,2 WITH DATA;
+NOTICE:  adding index _materialized_hypertable_12_device_day_idx ON _timescaledb_internal._materialized_hypertable_12 USING BTREE(device, day)
+NOTICE:  refreshing continuous aggregate
+SELECT * FROM weekly_temp_no_refresh;
+ day | device | avg_temp 
+-----+--------+----------
+(0 rows)
+
+SELECT * FROM weekly_temp_refresh;
+             day              | device |     avg_temp     
+------------------------------+--------+------------------
+ Sun Apr 26 17:00:00 2020 PDT |      3 | 21.5631067961165
+ Sun Apr 26 17:00:00 2020 PDT |      1 | 17.2474226804124
+ Sun May 03 17:00:00 2020 PDT |      0 | 16.7659574468085
+ Sun May 03 17:00:00 2020 PDT |      1 | 22.7272727272727
+ Sun Apr 26 17:00:00 2020 PDT |      0 | 17.8181818181818
+ Sun Apr 26 17:00:00 2020 PDT |      2 | 18.9803921568627
+ Sun May 03 17:00:00 2020 PDT |      2 |  15.811320754717
+ Sun May 03 17:00:00 2020 PDT |      3 |               19
+(8 rows)
+
+CALL refresh_continuous_aggregate('weekly_temp_no_refresh', NULL, NULL);
+SELECT * FROM weekly_temp_no_refresh;
+             day              | device |     avg_temp     
+------------------------------+--------+------------------
+ Sun Apr 26 17:00:00 2020 PDT |      3 | 21.5631067961165
+ Sun Apr 26 17:00:00 2020 PDT |      1 | 17.2474226804124
+ Sun May 03 17:00:00 2020 PDT |      0 | 16.7659574468085
+ Sun May 03 17:00:00 2020 PDT |      1 | 22.7272727272727
+ Sun Apr 26 17:00:00 2020 PDT |      0 | 17.8181818181818
+ Sun Apr 26 17:00:00 2020 PDT |      2 | 18.9803921568627
+ Sun May 03 17:00:00 2020 PDT |      2 |  15.811320754717
+ Sun May 03 17:00:00 2020 PDT |      3 |               19
+(8 rows)
+
+SELECT * FROM weekly_temp_refresh;
```

Comment:
This output seems unnecessary given that the some output is shown before `weekly_temp_no_refresh`.

I suggest removing this or replacing with the aggregate query directly on the hypertable.

Reply:
This is only for verifying that `WITH NO DATA` does not refresh and that `WITH DATA` does a refresh. I do not think there is any reason to compare this with the hypertable since there are plenty of tests that check that continuous aggregates work as expected.

Reply:
Removed the second query of the non-refreshed view.